temporalio/worker/_workflow.py

```diff
@@ -241,8 +257,25 @@ async def _handle_activation(
         # Always set the run ID on the completion
         completion.run_id = act.run_id
 
-        # Encode the completion if there's a codec
-        if self._data_converter.payload_codec:
+        # If there is a remove-from-cache job, do so
+        if cache_remove_job:
+            if act.run_id in self._running_workflows:
+                logger.debug(
+                    "Evicting workflow with run ID %s, message: %s",
+                    act.run_id,
+                    cache_remove_job.message,
+                )
+                del self._running_workflows[act.run_id]
+            else:
+                logger.debug(
```

Comment:
Consider making this one a warn. Just like the "cache already exists", falling in this case is most probably a symptom of something incorrect elsewhere, which we're unlikely to ever find out if its only pronted out as debug.

Reply:
This isn't new code, I just moved it from below. But I can update this log level.

Reply:
Updated